Frontend/CarRentalAdmin/src/app/components/login/login.component.html

```diff
@@ -0,0 +1,72 @@
+<div class="col-md-3 mx-auto">
+    <div class="card card-lg rounded-1">
+        <div class="card-header">
+            <h3>Enter your credentials</h3>
+        </div>
+        <div class="card-body">
+            <div *ngIf="authFailed" class="alert alert-danger">
+                {{authFailedMessage}}
+            </div>
+            <form [formGroup]="loginForm" (ngSubmit)="onSubmit()">
+                <div class="form-group">
+                    <label for="username">Username</label>
+                    <input
+                        required
+                        class="form-control"
+                        id="username"
+                        type="text"
+                        formControlName="username"
+                        (change)="resetAuthStatus()"
+                    >
+                    <div
+                        *ngIf="checkIfErrorsOccurred(loginForm.controls.username)"
+                        class="alert alert-danger"
+                    >
+                        <div *ngIf="loginForm.controls.username.errors?.['required']">
```

Comment:
to avoid duplication of such construction you can write function that accepts control name (you can get control by name from form group) and error type
or just function that returns control by name and call .hasError on it with error type as param